### DISCUSSION TITLE
What is the right approach to display an entity creational dialog/popup?

### DISCUSSION THREAD

**Discussion by xperiandri** [unoplatform/uno.extensions#2237] [Q&A] [answered]
I have a bit complicated entity creational dialog
![image](https://github.com/unoplatform/uno.extensions/assets/2365592/56a690d7-4cd4-4db3-b927-3f1cda552aa7)

How should I implement it with navigation extensions and `INavigator`?

**Comment by kazo0** [comment, marked as answer]
@nickrandolph might have some insights but I imagine you would want to navigate to this page using the `Dialog` qualifier, as seen in this guide: https://platform.uno/docs/articles/external/uno.extensions/doc/Learn/Navigation/HowTo-ShowFlyout.html

And I'm also thinking you'd want to navigate to this page and wait for a result back with the created entity or the cancellation. Which can be accomplished with the `INavigator` using the `NavigateForResultAsync` extension method. This is a guide for requesting data through navigation: https://platform.uno/docs/articles/external/uno.extensions/doc/Learn/Navigation/HowTo-SelectValue.html. Although this one is using the `GetDataAsync` method instead and I'm a little fuzzy on which would be more appropriate for this scenario.

You can see a usage of either in our Commerce sample app [here](https://github.com/unoplatform/Uno.Samples/blob/abf8042e9cf5eb9057ca7a4fa4476a7a657e70fd/reference/Commerce/Commerce/Commerce/Presentation/ShellViewModel.cs#L39-L48)

**Reply by nickrandolph** [reply]
If you inherit from contentdialog or flyout you don't need the Dialog qualifier as Navigation will automatically display it correctly. If you inherit from Page or UserContorl and specify Dialog, Navigation will put your content within a Flyout (which may not be the desired outcome as it's dismissable)

**Reply by xperiandri** [reply]
Works great! Thanks!
